memgpt/orm/tool.py

```diff
@@ -46,16 +46,16 @@ class Tool(SqlalchemyBase, OrganizationMixin):
     organization: Mapped["Organization"] = relationship("Organization", back_populates="tools", lazy="selectin")
 
     @classmethod
-    def read(cls, db_session: "Session", name: str) -> "Tool":
+    def read_by_name(cls, db_session: "Session", name: str) -> "Tool":
         if found := db_session.query(cls).filter(cls.name == name, cls.is_deleted == False).scalar():
             return found
         raise NoResultFound(f"{cls.__name__} with name {name} not found")
 
-    @classmethod
-    def read_by_id(cls, db_session: "Session", id: str) -> "Tool":
-        if found := db_session.query(cls).filter(cls.id == id, cls.is_deleted == False).scalar():
-            return found
-        raise NoResultFound(f"{cls.__name__} with id {id} not found")
+    # @classmethod
```

Comment:
I removed this because it seems duplicated with the parent class `read`, and I think its confusing to pretend that `Tool.name` is the id

Reply:
if you want to be explicit about it without adding extra one-off methods you could also overload the parent read on Tool to take a `name` param instead. 
This is one of those things that functional programmers hate but overloading (extending the arity) this way is an OOP pattern as opposed to lots of single-use getters